app/services/optIn/OptInService.scala

```diff
@@ -42,6 +48,115 @@ class OptInService @Inject()(itsaStatusUpdateConnector: ITSAStatusUpdateConnecto
       getOrElse(true)//todo this should default to false, set to true until journey setup is done correctly
   }
 
-  def availableOptInTaxYear() = List(TaxYear.forYearEnd(2023), TaxYear.forYearEnd(2024)) //todo TBD
+  def availableOptInTaxYear()(implicit user: MtdItUser[_],
+                              hc: HeaderCarrier,
+                              ec: ExecutionContext): Future[Seq[TaxYear]] = fetchOptInProposition().map(_.availableOptInYears.map(_.taxYear))
 
+  def fetchSavedOptInSessionData()(implicit user: MtdItUser[_],
+                                   hc: HeaderCarrier,
+                                   ec: ExecutionContext): Future[Option[OptInSessionData]] = {
+
+    val savedOptInSessionData = for {
+      sessionData <- OptionT(repository.get(hc.sessionId.get.value, OptInJourney.Name))
+      optInSessionData <- OptionT(Future.successful(sessionData.optInSessionData))
+    } yield optInSessionData
+
+    savedOptInSessionData.value
+  }
+
+  def fetchSavedOptInProposition()(implicit user: MtdItUser[_],
+                              hc: HeaderCarrier,
+                              ec: ExecutionContext): Future[Option[OptInProposition]] = {
+
+    val savedOptInProposition = for {
+
+      optInSessionData <- OptionT(fetchSavedOptInSessionData())
+      contextData <- OptionT(Future.successful(optInSessionData.optInContextData))
```

Comment:
It's only a small improvement, but you might like to consider using `OptionT.fromOption[Future](...)` in place of `OptionT(Future.successful(...))` for these 3 occurrences.

Reply:
done